repair/repair.cc

```diff
@@ -1445,7 +1445,9 @@ future<> repair_service::bootstrap_with_repair(locator::token_metadata_ptr tmptr
                         auto old_endpoints_in_local_dc = get_old_endpoints_in_local_dc();
                         auto rf_in_local_dc = get_rf_in_local_dc();
                         if (everywhere_topology) {
-                            neighbors = old_endpoints_in_local_dc;
+                            neighbors = old_endpoints_in_local_dc.empty() ? old_endpoints : old_endpoints_in_local_dc;
+                            rlogger.debug("bootstrap_with_repair: keyspace={}, range={}, old_endpoints={}, new_endpoints={}, old_endpoints_in_local_dc={}, neighbors={}",
+                                    keyspace_name, desired_range, old_endpoints, new_endpoints, old_endpoints_in_local_dc, neighbors);
```

Comment:
Shouldn't it sync with a quorum?

consider: 
1. dc1=(n1, n2, n3); a mutation is in n2, n3 but not in n1
2. bootstrap dc2:n4, picks n1, mutation not streamed
3. boostrap dc2:(n5, n6), both stream from n1
4. dc3 does not have the mutation

So, new nodes should stream from a quorum.

Reply:
This is everywhere strategy, in your example, n4 will pick to sync with the old nodes , n1, n2, n3, so n4 will have the mutation. n5 and n6 will sync from n4. So all nodes in dc2 will have the mutation.

Reply:
I see.

Everywhere is pretty poor, it's easy for a node to lose synchronization and then have to wait for repair. We should switch all such tables to a new Everywhere that uses Raft, so it auto-repairs.

Reply:
Yes, CDC is the current user of Everywhere. It is problematic in the case where we want to ignore dead nodes when bootstrap. We do not have the semantics to do cql writes with Everywhere while ignoring some nodes. Hope we can move away from it at some point.

For the external usage of Everywhere (user created tables), I am not sure if it is useful in practice.

Reply:
CDC always writes with CL=ALL to its Everywhere table.

But regular tables can also be created with Everywhere strategy and they are documented. We should correctly handle it in also with CL=QUORUM etc.

Reply:
Do you know a user case where Everywhere + CL QUORUM would be helpful?

Reply:
Nothing specific. I imagine if someone wanted to store a small amount of metadata that's rarely updated, but wants it to be very durable, but at the same time with reasonable availability, they could use Everywhere with CL=QUORUM.

Reply:
In this case, the user can use the normal network strategy with a higher constant RF.  The property given by the everwhere strategy RF == number of nodes, is not useful. At some point, the writes needs X replicas. Later, after nodes are added or removed, the writes needs Y replicas.

Reply:
The property is useful. It would be a huge pain in the... to modify the RF every time we bootstrap/remove a node. I'm glad we don't have to do this in CDC code, for example.